modules/layers/src/icon-layer/icon-manager.js

```diff
@@ -12,6 +14,19 @@ const DEFAULT_TEXTURE_MAG_FILTER = GL.LINEAR;
 
 const noop = () => {};
 
+/**
+ * {
+ *   [gl]: {
+ *     texture, // Texture2D instance containing the icons
+ *     mapping, // Object, {[id]: {width, height, }}
+ *     xOffset, // right position of last icon
+ *     yOffset, // top position of last icon
+ *     canvasHeight // canvas height
+ *   }
+ * }
+ */
+const cached = new LRUCache(3);
```

Comment:
Comparing to the text layer, it is less significant for icon layers to be able to share textures with each other. This is also a potential source of bugs if several icon layers try to use the same icon id but different images.